handler/data_handler.go

```diff
@@ -40,6 +40,14 @@ func (postgresDataHandler *PostgresDataHandler) HandleEntryBatch(batchedEntries
 
 	// Get the correct db handle.
 	dbHandle := postgresDataHandler.GetDbHandle()
+	if err = AcquireAdvisoryLock(dbHandle); err != nil {
```

Comment:
I would remove this lock - this will cause our multi-threaded hypersync to become single threaded - a better spot to throw this would be right before we initialize a transaction and right after we commit or roll it back - transactions are only run during block sync mode, where we know we're single threaded.